src/BSeries.jl

```diff
@@ -2024,8 +2064,15 @@ function _is_energy_preserving_sparse(trees, coefficients)
 
     # The components of `energy_preserving_basis` are the columns of the matrix `M`.
     M = sparse(rows, cols, vals, length(coefficients), maximum(cols))
-    rank_M = rank(M)
+    # Compute the singular value decomposition (SVD) of the sparse matrix
+    U, Σ, V = svd(Matrix(M))
```

Comment:
No. We explicitly used sparse matrices for efficiency. You convert them here to dense matrices. That defeats the purpose of this way of computing things and will likely make it quite a lot slower for bigger systems.